url542ae48576_tonyheadford_m68k_tgz-pJ8-m68k_MonitorJ8/src/m68k/Monitor.java

```diff
@@ -0,0 +1,966 @@
+						return;
+					}
+					int v = parseInt(value);
+					cpu.writeMemoryLong(addr, v);
+				}
+				catch(NumberFormatException e)
+				{
+					e.printStackTrace();
+				}
+			}
+		}
+	}
+
+	protected void handleMemWord(String[] tokens)
+	{
+		if(tokens.length != 2 && tokens.length != 3)
+		{
+			writer.println("usage: mw <address> [value]");
+		}
+		else
+		{
+			String address = tokens[1];
+
+			if(tokens.length == 2)
+			{
+				// just read address
+				try
+				{
+					int addr = parseInt(address);
+					if(addr < 0 || addr >= memory.size())
+					{
+						writer.println("Address out of range");
+						return;
+					}
+					writer.printf("%08x  %04x\n", addr, cpu.readMemoryWord(addr));
+				}
+				catch(NumberFormatException e)
+				{
+					e.printStackTrace();
+				}
+			}
+			else
+			{
+				//write address
+				String value = tokens[2];
+				try
+				{
+					int addr = parseInt(address);
+					if(addr < 0 || addr >= memory.size())
+					{
+						writer.println("Address out of range");
+						return;
+					}
+					int v = parseInt(value);
+					cpu.writeMemoryWord(addr, v);
+				}
+				catch(NumberFormatException e)
+				{
+					e.printStackTrace();
+				}
+			}
+		}
+	}
+	protected void handleMemByte(String[] tokens)
+	{
+		if(tokens.length != 2 && tokens.length != 3)
+		{
+			writer.println("usage: mb <address> [value]");
+		}
+		else
+		{
+			String address = tokens[1];
+
+			if(tokens.length == 2)
+			{
+				// just read address
+				try
+				{
+					int addr = parseInt(address);
+					if(addr < 0 || addr >= memory.size())
+					{
+						writer.println("Address out of range");
+					}
+					else
+					{
+						writer.printf("%08x  %02x\n", addr, cpu.readMemoryByte(addr));
+					}
+				}
+				catch(NumberFormatException e)
+				{
+					e.printStackTrace();
+				}
+			}
+			else
+			{
+				//write address
+				String value = tokens[2];
+				try
+				{
+					int addr = parseInt(address);
+					if(addr < 0 || addr >= memory.size())
+					{
+						writer.println("Address out of range");
+						return;
+					}
+					int v = parseInt(value);
+					cpu.writeMemoryByte(addr, v);
+				}
+				catch(NumberFormatException e)
+				{
+					e.printStackTrace();
+				}
+			}
+		}
+	}
+
+	protected void handleLoad(String[] tokens)
+	{
+		// load address file
+		if(tokens.length != 3)
+		{
+			writer.println("usage: load <address> <file>");
+			return;
+		}
+
+		int address;
+		try
+		{
+			address = parseInt(tokens[1]);
+		}
+		catch(NumberFormatException e)
+		{
+			writer.println("Invalid address specified [" + tokens[1] + "]");
+			return;
+		}
+
+		File f = new File(tokens[2]);
+		if(!f.exists())
+		{
+			writer.println("Cannot find file [" + tokens[2] + "]");
+			return;
+		}
+
+		if(address + (int)f.length() >= memory.size())
+		{
+			//need larger memory
+			writer.println("Need larger memory to load this file at " + tokens[1]);
+			return;
+		}
+		try
+		{
+			FileInputStream fis = new FileInputStream(f);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fis*. The resource is referenced at line 921. The resource is closed at line 922. There are other execution paths that do not contain closure statements, for example, when FileInputStream.read throws an exception. Ensure *fis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).